src/renderer/components/+workloads/rollout-menu.tsx

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2021 OpenLens Authors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+import React from "react";
+import type { WorkloadKubeObject } from "../../api/workload-kube-object";
+import { MenuItem } from "../menu";
+import { ConfirmDialog } from "../confirm-dialog";
+import { Notifications } from "../notifications";
+import { Icon } from "../icon";
+import { ScaleDialog } from "./scale-dialog";
+import { apiManager } from "../../api/api-manager";
+
+interface Props<T extends WorkloadKubeObject> {
+  workloadKubeObject: T;
+  interactive?: boolean;
+}
+
+export function renderRestart<T extends WorkloadKubeObject>(workloadKubeObject: T, api: any, interactive?: boolean) {
+
+  if (!workloadKubeObject) {
+    return null;
+  }
+
+  return (
+    <MenuItem onClick={() => ConfirmDialog.open({
+      ok: async () => {
+        try {
+          await api.restart({
+            namespace: workloadKubeObject.getNs(),
+            name: workloadKubeObject.getName(),
+          });
+        } catch (err) {
+          Notifications.error(err);
+        }
+      },
+      labelOk: `Restart`,
+      message: (
+        <p>
+          Are you sure you want to restart {workloadKubeObject.kind} <b>{workloadKubeObject.getName()}</b>?
+        </p>
+      ),
+    })}>
+      <Icon material="autorenew" tooltip="Restart" interactive={interactive}/>
+      <span className="title">Restart</span>
+    </MenuItem>
+  );
+}
```

Comment:
This should be moved to separate component as all of the rollout commands will share exact same code.